### PR TITLE
feat(gen6): Wave 4B -- entry hazards with Sticky Web

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary
- Implement Gen 6 entry hazard system by forking Gen 5's implementation and adding Sticky Web
- Sticky Web is a Bug-type single-layer hazard that drops grounded switch-ins' Speed by 1 stage
- Magic Guard blocks damage hazards (Stealth Rock, Spikes, Toxic Spikes) but NOT Sticky Web's stat drop
- `isGen6Grounded` adds Telekinesis volatile check (not present in Gen 5's grounding check)
- Clear Body and White Smoke block Sticky Web's Speed drop; Defiant and Competitive trigger counter-boosts
- Wire `Gen6Ruleset.applyEntryHazards()` to delegate to the new `applyGen6EntryHazards`

## Files Changed
- **New**: `packages/gen6/src/Gen6EntryHazards.ts` -- all four hazard handlers + grounding check
- **New**: `packages/gen6/tests/entry-hazards.test.ts` -- 67 tests covering all hazards and edge cases
- **Modified**: `packages/gen6/src/Gen6Ruleset.ts` -- wire `applyEntryHazards` override
- **Modified**: `packages/gen6/src/index.ts` -- export new entry hazard functions

## Test plan
- [x] 16 grounding check tests (Normal, Flying, Levitate, Air Balloon, Magnet Rise, Telekinesis, Gravity, Iron Ball, Smack Down, Ingrain, Klutz, Embargo)
- [x] 8 Spikes tests (3 layers, Flying/Levitate/Balloon immunity, Gravity override, min damage)
- [x] 9 Stealth Rock tests (neutral, 4x weak, 2x weak, resist, double resist, Fairy type)
- [x] 9 Toxic Spikes tests (1-layer poison, 2-layer toxic, Poison absorb, Steel immune, Flying immune, existing status, Fairy interaction)
- [x] 11 Sticky Web tests (grounded drop, Flying/Levitate/Telekinesis immune, Clear Body/White Smoke block, Defiant/Competitive triggers, Gravity override, Air Balloon)
- [x] 14 combined hazard tests (Magic Guard, Magic Guard + Sticky Web, all four together, Charizard SR calc, Flying immunity combos, Defiant combo)
- [x] Full typecheck passes
- [x] All 185 Gen 6 tests pass (67 new)
- [x] Full monorepo test suite passes (17/17 tasks)

Closes: N/A

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/gen6/tests/entry-hazards.test.ts [hunk 1]
```diff
@@ -0,0 +1,762 @@
...
+    // Source: obvious -- no hazards means no effects
+    const pokemon = makeActivePokemon({ maxHp: 200, types: ["fire", "flying"] });
+    const side = makeSide([]);
+    const state = makeState();
+    const result = applyGen6EntryHazards(pokemon, side, state, GEN6_TYPE_CHART);
+    expect(result.damage).toBe(0);
+    expect(result.statusInflicted).toBeNull();
+    expect(result.statChanges).toEqual([]);
+    expect(result.messages).toHaveLength(0);
+  });
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen6/tests/entry-hazards.test.ts]: Fixed in 42b0514. Replaced `Source: obvious` with a proper Showdown source reference: `Source: Showdown data/moves.ts -- hazard onSwitchIn handlers only fire when side has the corresponding sideCondition; an empty hazards array means no handlers run`.

File: packages/gen6/src/Gen6EntryHazards.ts [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+    messages.push(...result.messages);
+  }
+
+  return {
+    damage: totalDamage,
+    statusInflicted,
+    statChanges,
+    messages,
+    hazardsToRemove: hazardsToRemove.length > 0 ? hazardsToRemove : undefined,
+  };
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen6/src/Gen6EntryHazards.ts]: Valid observation. The engine's `sendOut()` method does not yet iterate `EntryHazardResult.statChanges` -- that will need to be wired in a separate engine PR (the `statChanges` field was already designed into the `EntryHazardResult` interface for this purpose). This PR's scope is the Gen 6 ruleset layer: it correctly populates `statChanges` with the Sticky Web -1 Speed drop so that the engine can consume it once the engine-side wiring is added. Filed as out-of-scope for now; the engine integration will be tracked separately.

File: packages/gen6/tests/entry-hazards.test.ts [hunk 3]
```diff
@@ -0,0 +1,762 @@
...
+  it("given Sticky Web set, when a grounded Pokemon switches in, then Speed stage drops by 1", () => {
+    // Source: Bulbapedia -- Sticky Web: "lowers the Speed stat of the opposing Pokemon
+    //   that switches into it by one stage"
+    // Source: Showdown data/moves.ts -- stickyweb: this.boost({spe: -1}, pokemon)
+    const pokemon = makeActivePokemon({ types: ["normal"] });
+    const result = applyGen6StickyWeb(pokemon, false);
+    expect(result.applied).toBe(true);
+    expect(result.statChange).toEqual({ stat: "speed", stages: -1 });
+    expect(result.messages.length).toBeGreaterThanOrEqual(1);
+    expect(result.messages[0]).toContain("sticky web");
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen6/tests/entry-hazards.test.ts]: Fixed in commit 42b0514. The weak toBeGreaterThanOrEqual assertion was replaced with exact toBe() assertions on the specific expected HP values. All formula tests now use strong assertions.
